rfps/rfp-filecoin-wallets.md

```diff
@@ -0,0 +1,159 @@
+
+> NOTE: Nodes have wallet CLIs and APIs that will be further developed to meet the Filecoin Spec (e.g. Multisig wallets). A reference tool for signing raw transactions outside of a node will also be created by the Filecoin team.
+
+The goal of an enhanced open source wallet are to:
+
+- provide the community with an open source implementation that has better UX for non-technical end-users
+- ensure good node API coverage for wallet functionality
+- help test parts of node APIs that interact with wallets
+- reveal additional architectural components of wallet infrastructure that could be open sourced
+- provide a reference or re-usable library for future application developers
+- provide additional features to improve usability for end-users or developers
+
+There are many types of wallets and many features that could be included. We leave the choice of which would greatly improve usability and adoption up to you.
+
+Possible wallet types and features:
+
+- Web, Mobile, Desktop (or Paper)
+- HD (Hierarchical Deterministic) wallets that support multiple addresses from a mnemonic seed
+- Secure private key management and storage
+- Wallets customized to support Mining
+- 2FA or Multi-device backup
+- MultiSig wallets
+- Air-gapped cold wallet vs. hot wallets
+- Support for BLS in addition to secp256k1
+- Cross-platform integration
+- Suggest additional features
+
+
+### Track 2: Wallet Tools and Integrations
+-------------------------------------------
+
+We are also seeking proposals for open source tools and integrations for wallets. This can include supporting libraries or infrastructure that provide new and existing wallets with important services for integrating with the Filecoin Network or enhanced usability.
+
+Some examples:
+
+- Hardware wallet integration (e.g. [lotus issue #39](https://github.com/filecoin-project/lotus/issues/39))
+- Browser integration or with other applications
+- Wallet frameworks for app developers
+- Gas estimation tools
+- Privacy features
+- Exchange or swap integration
+- Spot pricing tools
+- Specialized Wallet API providers
+- Suggest additional tools and services
+
+##  Deliverables
+
+An open-sourced, dual-licensed (under MIT and APACHE2) implementation that:
+
+for Track 1:
+
+- enables users to receive and transfer FIL
+- supports secure key management, backup and recovery
+- shows transactions confirmed by the network
+- supports any additional features listed above
+
+for Track 2:
+
+- provides useful features or services to wallet users and/or developers
+- is easy to integrate with new and existing wallets
+- would help grow adoption or improve security
+
+for both Tracks:
+
+- has a well-documented API
+- works with a Filecoin node API (local or remote) to perform the intended workflow
+- has tutorials with demo usage
+- is written in a mainstream language to ensure wider adoption, future interoperability and re-use
```

Comment:
What are examples of the mainstream languages you're thinking of? Can you provide some examples in the form `(e.g. Go, Python, etc)`?

Reply:
Can we review their choice of mainstream language at the Review stage? I don't want to list out all languages I consider mainstream here, at some point it starts to be a discussion.

Reply:
Sure